pages/index.tsx

```diff
@@ -1,64 +1,83 @@
-import { ChangeEvent, useEffect, useState } from "react";
-import { CrossCircleSVG, Spinner, Tag } from "@ensdomains/thorin";
+import { ChangeEvent, useState } from "react";
+import { CrossCircleSVG } from "@ensdomains/thorin";
 import { HomepageBg } from "@/components/01-atoms";
 import { isNameAvailable } from "@/lib/utils/blockchain-txs";
 import { ENSName, buildENSName } from "@namehash/ens-utils";
 import { DebounceInput } from "react-debounce-input";
 import { useRouter } from "next/router";
 import { normalize } from "viem/ens";
 import Link from "next/link";
+import { EnsDomainStatus } from "@/types/ensDomainStatus";
+import { domainWithEth, hasMoreThanOneDot } from "@/lib/utils/formats";
+import { EnsDomainStatusComponents } from "@/components/02-molecules";
 
 export default function Home() {
   const router = useRouter();
   const [domain, setDomain] = useState("");
-  const [isAvailable, setIsAvailable] = useState<boolean | null>(null);
-  const [domainError, setDomainError] = useState(false);
+  const [domainStatus, setDomainStatus] = useState<EnsDomainStatus>(
+    EnsDomainStatus.Available
+  );
 
-  useEffect(() => {
-    if (!domain) return;
+  const updateDomainStatus = (searchedDomain: string) => {
+    if (!searchedDomain) return EnsDomainStatus.Invalid;
 
-    setIsAvailable(null);
+    setDomainStatus(EnsDomainStatus.Searching);
 
     let ensName: null | ENSName = null;
+
+    // check if the domain is valid
     try {
-      ensName = buildENSName(domain);
+      normalize(searchedDomain);
+      ensName = buildENSName(searchedDomain);
     } catch (error) {
       console.error(error);
-      setIsAvailable(false);
+      setDomainStatus(EnsDomainStatus.Invalid);
       return;
     }
 
+    // check if domain is supported
+    if (hasMoreThanOneDot(domainWithEth(searchedDomain))) {
+      setDomainStatus(EnsDomainStatus.NotSupported);
+      return;
+    }
+
+    // check if domain is available
     isNameAvailable(ensName)
```

Comment:
What about transforming this into a new try catch block just as below?

Await can be used, then 👍🏼